frontend/public/components/dashboard/dashboards-page/overview-dashboard/utilization-card.tsx

```diff
@@ -117,135 +120,112 @@ const getQueries = (flags: FlagsObject) => {
-    prometheusResults,
-    queries[OverviewQuery.POD_UTILIZATION].utilization,
-    UTILIZATION_QUERY_HOUR_MAP[duration],
-  );
-  const [networkUtilization, networkUtilizationError] = getPrometheusQueryResponse(
-    prometheusResults,
-    queries[OverviewQuery.NETWORK_UTILIZATION].utilization,
-    UTILIZATION_QUERY_HOUR_MAP[duration],
-  );
-  const [networkTotal, networkTotalError] = getPrometheusQueryResponse(
-    prometheusResults,
-    queries[OverviewQuery.NETWORK_UTILIZATION].total,
-  );
+    utilizationQuery,
+    totalQuery,
+    duration,
+    title,
+    TopConsumerPopover,
+    humanizeValue,
+    byteDataType,
+    setTimestamps,
+  }) => {
+    React.useEffect(() => {
+      watchPrometheus(utilizationQuery, null, UTILIZATION_QUERY_HOUR_MAP[duration]);
+      totalQuery && watchPrometheus(totalQuery);
+      return () => {
+        stopWatchPrometheusQuery(utilizationQuery, UTILIZATION_QUERY_HOUR_MAP[duration]);
+        totalQuery && stopWatchPrometheusQuery(totalQuery);
+      };
+    }, [watchPrometheus, stopWatchPrometheusQuery, duration, utilizationQuery, totalQuery]);
+    const [utilization, utilizationError] = getPrometheusQueryResponse(
+      prometheusResults,
+      utilizationQuery,
+      UTILIZATION_QUERY_HOUR_MAP[duration],
+    );
+    const [total, totalError] = getPrometheusQueryResponse(prometheusResults, totalQuery);
 
-  const cpuStats = getRangeVectorStats(cpuUtilization);
-  const cpuMax = getInstantVectorStats(cpuTotal);
-  const memoryStats = getRangeVectorStats(memoryUtilization);
-  const memoryMax = getInstantVectorStats(memoryTotal);
-  const storageStats = getRangeVectorStats(storageUtilization);
-  const storageMax = getInstantVectorStats(storageTotal);
-  const networkStats = getRangeVectorStats(networkUtilization);
-  const networkMax = getInstantVectorStats(networkTotal);
-  const podStats = getRangeVectorStats(podUtilization);
+    const stats = getRangeVectorStats(utilization);
+    const max = getInstantVectorStats(total);
+
+    setTimestamps && setTimestamps(stats.map((stat) => stat.x as Date));
+
+    return (
+      <UtilizationItem
+        title={title}
+        data={stats}
+        error={utilizationError || totalError}
+        isLoading={!utilization || !total}
+        humanizeValue={humanizeValue}
+        byteDataType={byteDataType}
+        query={utilizationQuery}
+        max={max.length ? max[0].y : null}
+        TopConsumerPopover={TopConsumerPopover}
+      />
+    );
+  },
+);
+
+export const UtilizationCard = connectToFlags(
+  ...plugins.registry.getRequiredFlags([isDashboardsOverviewUtilizationItem]),
+)(({ flags }) => {
+  const queries = React.useMemo(() => getQueries(flags), [flags]);
+  const [timestamps, setTimestamps] = React.useState<Date[]>();
+  const [duration, setDuration] = React.useState(metricDurations[0]);
 
   const cpuPopover = React.useCallback(
-    ({ current }) => (
+    React.memo<TopConsumerPopoverProp>(({ current }) => (
```

Comment:
Again. Just for readability - move the `memo()` call up and use well-named variable for it, please.